### DISCUSSION TITLE
Internal Checksum Error Mega Drives games

### DISCUSSION THREAD

**Discussion by Mythris29** [sanni/cartreader#426] [Need help]
Sorry if this has been asked before but so far SNES, Gameboy and N64 games have dumped no issues with my v5 cart reader. When I try Genesis/Mega Drive games the 3 games I own give me an Internal Checksum...Error. I've tried cleaning without effect. Any other suggestions or do I maybe have a bad soldering job somewhere on that connector?

I'm running v9.1 firmware with the updated console.txt files.

**Comment by VincentEmond** [comment]
I am guessing you already did that but just in case make sure you are in 5V mode. 

- I don't know if it's just me but my Genesis games seems to degrade faster and be in overall worst shape than my SNES games. You could try another round of cleaning the contacts with Isopropyl alcohol.

- I find that pulling the cart a little bit so that it's not inserted completely down the slot but still making contact sometime gives me better results with games in bad shape.

- Do the game work in your console?

- If I remember correctly I think there's no hard requirement that the internal checksum in the ROM match the calculated checksum. I'm pretty sure the checksum was just there for debugging and troubleshooting and not checked by the sega console. I seem to remember dumping a sega game that had a bad checksum and still came up with a verified hash on nointro. It's a bit odd that all your games do that tho. Do you end up with a file on your sd card or does it crash before dumping?

- If you have the files. Calculate their sha-1 hash and check the nointro database for a match.

- Again if you have the files try opening them in an Hex editor and take a quick look at the data. If everything is 00 or FF or you got absolutely no ASCII text that make sense then there's definitely a problem.

- Check your solder joints, make sure there are no solder bridges, cold solder joints, or pins with too much or too little solder.

- Make sure there's nothing metallic in the cartridge slot causing a short between pins.

- If you have a multimeter with a continuity function you can easily test for unwanted solder bridges or shorted pins.

- Last resort scenario but I bought a sega cd ram cartridge at some point and my genesis wouldn't recognize it no matter what. So what I did is put a little bit of fresh solder on the cartridge pads. And I have to stress "a little bit" if the fresh solder is too thick it won't fit into the cartridge slot and you could damage your console and/or the game cartridge. Try that at your own risk.

I am curious what games are you trying to dump?

**Reply by Mythris29** [reply]
Thank you for replying these suggestions were exactly what I was looking for. Now for the results...

> I am guessing you already did that but just in case make sure you are in 5V mode.

 Yes, confirmed the switch is in the 5v position. 

> * I don't know if it's just me but my Genesis games seems to degrade faster and be in overall worst shape than my SNES games. You could try another round of cleaning the contacts with Isopropyl alcohol.

I went ahead and brought out the brass polish I've used on NES games that have needed extra cleaning (with sucess) cleaned it with that then washed it off with ISO again. No change

> * I find that pulling the cart a little bit so that it's not inserted completely down the slot but still making contact sometime gives me better results with games in bad shape.

I tried this but after a few positions still no change (or didn't read at all)

> * Do the game work in your console?

Yes but it's a retron5 HD not original genesis hardware in case that matters

> * If I remember correctly I think there's no hard requirement that the internal checksum in the ROM match the calculated checksum. I'm pretty sure the checksum was just there for debugging and troubleshooting and not checked by the sega console. I seem to remember dumping a sega game that had a bad checksum and still came up with a verified hash on nointro. It's a bit odd that all your games do that tho. Do you end up with a file on your sd card or does it crash before dumping?

I do end up with a file after dumping.

> * If you have the files. Calculate their sha-1 hash and check the nointro database for a match.

Doesn't look like it matches with anything in the database. I also tried to run the dump in an emulator without success. 

> * Again if you have the files try opening them in an Hex editor and take a quick look at the data. If everything is 00 or FF or you got absolutely no ASCII text that make sense then there's definitely a problem.

Lots of 00 and ff at the beginning of the file but then there seems to be data and I recognized the game title in ascii

> * Check your solder joints, make sure there are no solder bridges, cold solder joints, or pins with too much or too little solder.

I went through and resoldered all of the pins on the slot, no change.

> * Make sure there's nothing metallic in the cartridge slot causing a short between pins.

Clear

> * If you have a multimeter with a continuity function you can easily test for unwanted solder bridges or shorted pins.

Clear on this too after checking with the multimeter (seems that the 15th pin from the right on the top is connected to the first and last pin on the top row but since they are so far apart I'm assuming that's on purpose)

> * Last resort scenario but I bought a sega cd ram cartridge at some point and my genesis wouldn't recognize it no matter what. So what I did is put a little bit of fresh solder on the cartridge pads. And I have to stress "a little bit" if the fresh solder is too thick it won't fit into the cartridge slot and you could damage your console and/or the game cartridge. Try that at your own risk.
> 
> I am curious what games are you trying to dump?

The game I'm focusing my efforts on is "World of Illusion starring Disney's Mickey Mouse & Donald Duck"

**Reply by VincentEmond** [reply]
- Well if it works on your retron then it means that at least the cartridge seem to be working correctly. 

- If it doesn't work in an emulator then it's probably bad. Maybe you can try in an another MD emulator but I doubt it would change anything.

- Were you able to dump a MD game successfully at least once before? Can you try with a different game?

- Do you at least get the right size for your dump (Mickey)?
1048576 bytes for both European and American game.

- Do you notice discrepancies in internal game title? For example internal title could be something `SONIC_THE_HEDGEHOG` but with bad contact it happened to me that I get something like `SBN3H_THE_UDGEHOG` in that case I know It'll result in a bad dump so I move the cartridge a bit and power cycle the reader. 

- Also make sure the cartridge is authentic and not a repro. I once made dumps of some GBA games and noticed that they wouldn't verify no matter what. Turns out I was scammed on Ebay and I got cheap repro cartridges instead of legit games. It's pretty easy to spot if you have the tools to open the cartridge. American MD/Genesis games are pretty easy to open but unfortunately Japanese games (to my knowledge) are not easily openable without damaging the label, I personally never risked it. I don't know about European MD cartridge as I don't own any.

**Comment by sanni** [comment]
If the CRC32 is correct and the ROM gets renamed to no-intro then the dump was successful. Some Megadrive games just lack the checksum or have a wrong checksum, or maybe calculate the checksum differently than the implemented standard way. 

If the CRC32 is wrong then either the game is a special case with more chips than a standard game and needs a special dumping function written for it or the contacts are indeed a little bit dirty. 
The Cart Reader is very picky when it comes to contact cleanliness, especially with GBA I have noticed I have to clean the cartridge until there is no black stuff on an alcohol soaked q-tip after whipping the contacts down.

**Reply by Mythris29** [reply]
I replied above if it helps answer some of your questions.

**Comment by Mythris29** [comment]
It looks like it was a bad socket and/or board. I tried a new board with a new Sega Genesis socket and it worked perfectly.
